pypesto/objective/amici_prediction.py

```diff
@@ -0,0 +1,439 @@
+import numpy as np
+import os
+import pandas as pd
+import h5py
+from typing import Sequence, Union, Callable, Tuple, Dict
+from warnings import warn
+from time import time
+
+from .constants import MODE_FUN
+from .amici import AmiciObjective
+
+
+try:
+    import amici
+    import amici.petab_objective
+    import amici.parameter_mapping
+    from amici.parameter_mapping import ParameterMapping
+except ImportError:
+    pass
+
+
+class PredictionConditionResult:
+    """
+    This class is a light-weight wrapper for the prediction of one simulation
+    condition of an amici model. It should provide a common api how amici
+    predictions should look like in pyPESTO.
```

Comment:
agreed, on could indeed move the result classes to another file. This may be helpful at a later point, have no strong opinion. Atm, this class is very simple, and it's also not planned to extend it too much... But true, would be nice to have that available also in application, which don't need amici...